tools/global_bss_check.py

```diff
@@ -0,0 +1,216 @@
+    parser.add_argument("mapFile", help="Path to a map file.")
+    parser.add_argument("mapFileExpected", help="Path to the expected map file. Optional, default is 'expected/mapFile'.", nargs="?", default="")
+    parser.add_argument("-a", "--print-all", help="Print all bss, not just non-matching.", action="store_true")
+    parser.add_argument("-n", "--no-fun-allowed", help="Remove amusing messages.", action="store_true")
+    args = parser.parse_args()
+
+    if args.mapFileExpected == "":
+        args.mapFileExpected = os.path.join("expected", args.mapFile)
+
+    badFiles, missingFiles, comparedDict = compareMapFiles(args.mapFile, args.mapFileExpected)
+    printCsv(badFiles, missingFiles, comparedDict, args.print_all)
+
+    if len(badFiles) + len(missingFiles) != 0:
+        print("", file=os.sys.stderr)
+
+    if len(badFiles) != 0:
+        print(colorama.Fore.RED + "  BAD" + colorama.Style.RESET_ALL)
+
+        for file in badFiles:
+            print(f"bss reordering in {file}", file=os.sys.stderr)
+        print("", file=os.sys.stderr)
+        
+        if not args.no_fun_allowed:
+            print(colorama.Fore.LIGHTWHITE_EX +
+            "  BSS is REORDERED!!\n"
+            "  Oh! MY GOD!!" 
+            + colorama.Style.RESET_ALL, file=os.sys.stderr)
+            print("", file=os.sys.stderr)
+
+    if len(missingFiles) != 0:
+        print(colorama.Fore.YELLOW + "  MISSING" + colorama.Style.RESET_ALL)
+
+        for file in missingFiles:
+            print(f"Symbols missing from {file}", file=os.sys.stderr)
+        print("", file=os.sys.stderr)
+
+        if not args.no_fun_allowed:
+            print(colorama.Fore.LIGHTWHITE_EX + "  Error, should (not) be in here " + colorama.Style.RESET_ALL, file=os.sys.stderr)
+            print("", file=os.sys.stderr)
+
+        print("Some files appear to be missing symbols. Have they been renamed or declared as static? You may need to remake 'expected'", file=os.sys.stderr)
+
+    if len(badFiles) + len(missingFiles) != 0:
+        return 1
+    
+    print("", file=os.sys.stderr)
+    print(colorama.Fore.GREEN + "  GOOD" + colorama.Style.RESET_ALL, file=os.sys.stderr)
+
+    if args.no_fun_allowed:
+        return 0
+
+    print("\n" + colorama.Fore.LIGHTWHITE_EX +
+    colorama.Back.RED + "                                  " + colorama.Back.RESET + "\n" +
+    colorama.Back.RED + "         CONGRATURATIONS!         " + colorama.Back.RESET + "\n" +
+    colorama.Back.RED + "    All Global BSS is correct.    " + colorama.Back.RESET + "\n" +
+    colorama.Back.RED + "             THANK YOU!           " + colorama.Back.RESET + "\n" +
+    colorama.Back.RED + "      You are great decomper!     " + colorama.Back.RESET + "\n" +
+    colorama.Back.RED + "                                  " + colorama.Style.RESET_ALL , file=os.sys.stderr)
+
+    return 0
+
+if __name__ == "__main__":
+    main()
```

Comment:
#re: ignored main return value
I guess you wanted to mimic the C behaviour of the main return value. You would need to do something like this to make it work.
```suggestion
if __name__ == "__main__":
    ret = main()
    exit(ret)
```